File: changer/src/main/kotlin/junny/land/xlsx/builder/extract/HeaderValues.java
package junny.land.xlsx.builder.extract;

import static java.util.Optional.ofNullable;

import java.lang.reflect.Field;
import java.util.ArrayList;
import junny.land.xlsx.annotation.Column;
import junny.land.xlsx.datas.HeaderDatas;

public class HeaderValues<T> {

    private final Class<T> clazz;

    public HeaderDatas extract() {
        ArrayList<String> headerList = new ArrayList<>();
        for (Field field : clazz.getDeclaredFields()) {
            String name = ofNullable(field.getAnnotation(Column.class))
                .map(Column::value)
                .orElseGet(field::getName);

            headerList.add(name);
        }
        return new HeaderDatas(headerList);
    }

    public HeaderValues(Class<T> clazz) {
        this.clazz = clazz;
    }
}

File: changer/src/main/kotlin/junny/land/xlsx/builder/extract/HeaderValues.kt
package junny.land.xlsx.builder.extract

import junny.land.xlsx.annotation.Column
import junny.land.xlsx.datas.HeaderDatas
import java.util.*

class HeaderValues<T>(
    private var clazz: Class<T>
) {

    fun extract(): HeaderDatas {
        val headerList = ArrayList<String>()
        for (field in clazz.declaredFields) {
            if (field.name == "this\$0") continue
            val name = Optional.ofNullable(field.getAnnotation(Column::class.java))
                .map(Column::value)
                .orElseGet { field.name }
            headerList.add(name)
        }
        return HeaderDatas(headerList)
    }
}